tools/nnyeah/nnyeah/MethodTransformations.cs

```diff
@@ -0,0 +1,834 @@
+using System;
+using System.Collections.Generic;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+#nullable enable
+
+namespace nnyeah {
+	public class MethodTransformations {
+		static List<Transformation>? allTransforms;
+		static Dictionary<string, Transformation>? transformTable;
+
+		public Dictionary<string, Transformation> GetTransforms (ModuleDefinition module, Func<List<bool>, CustomAttribute> attrBuilder)
+		{
+			if (transformTable is not null) {
+				return transformTable;
+			}
+
+			// there are two types of transforms here: ones that can be made statically
+			// and ones that need more in the way of state and intermediate variables.
+
+			var singleBool = new List<bool> () { true };
+
+			allTransforms = new List<Transformation> (transforms.Length);
```

Comment:
We normally make member variables upper case, I was confused here looking for a local transforms variable when it is defined outside of this function.

Maybe rename it to `BasicTransforms` or `StatelessTransforms`?